contract/views/res_partner_view.xml

```diff
@@ -19,7 +19,7 @@
                     groups="account.group_account_invoice"
                     icon="fa-book"
                     context="{'default_contract_type': 'sale', 'contract_type': 'sale'}"
-                    attrs="{'invisible': [('customer_rank','=',0)]}"
+                    attrs="{'invisible': [('sale_contract_count','=',0)]}"
```

Comment:
With this change the contracts button will not appear when creating a new customer.

When creating a customer from, for example, Invoicing > Customers the customer created is already assigned a customer_rank value, so I don't understand the problem with the current behaviour.

Reply:
In our use case, we use the Members module, and contacts are created from this module. Thank you for the comment!

Reply:
I would say, this is semantically more correct as the context is contract oriented. So, having contracts to show the button is better than 'being a customer'.

BUT, I'm wondering if showing it even in no contracts is better or not as with that feature, user can create a new contract for the partner directly. What do you think of ?

Reply:
A fairly common flow is to create the customer from the Invoicing > Customers option and once created click the smartbutton to create a new contract.
Any changes that involve losing the most basic flow to solve data integrity problems or "incorrect" processes by the user should not be included in the "base" modules.

Reply:
> A fairly common flow is to create the customer from the Invoicing > Customers option and once created click the smartbutton to create a new contract.

customer_rank == 0 if you create it from `Contacts` menu. So, this is not allowed in that case. Removing at all the domain is fair enough IMHO.